components/skin/ui/src/main/resources/PhenoTips/DocumentMeta.xml

```diff
@@ -37,8 +37,30 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{velocity}}{{html clean=false wiki=false}}
+  <content>{{velocity output=false}}
+#macro (__buildTabs $extName)
+  #foreach ($e in $services.uix.getExtensions($extName, {'sortByParameter' : 'order'}))
+    #set ($parameters = $e.getParameters())
+    #if ($parameters.get('enabled') != 'false')
+      #set ($label = "$!{parameters.get('label')}")
+      #set ($href = "$!{parameters.get('href')}")
+      #set ($cssId = "$!{parameters.get('cssId')}")
+      #if ($label != "" &amp;&amp; $href != "" &amp;&amp; $cssId != "")
+        &lt;div class="title-tab" id="$!{cssId}"&gt;
+          &lt;a href="$href"&gt;$label&lt;/a&gt;
+          &lt;h1 class="hidden"&gt;$label&lt;/h1&gt;
+        &lt;/div&gt;
+      #end
+    #end
+  #end
+#end
+{{/velocity}}
+
+{{velocity}}{{html clean=false wiki=false}}
 #if ($doc.getObject('PhenoTips.PatientClass') || $doc.getObject('PhenoTips.FamilyClass'))
+  #__buildTabs('org.phenotips.entitySheet.tab')
```

Comment:
I don't think we can do it now, but just wondering if there is a way to detect if XXXXClass is a class representing an entity (instead of hard-coding PatientClass + FamilyClass)? (@sdumitriu?)

Reply:
https://github.com/phenotips/phenotips/blob/master/components/entities/api/src/main/java/org/phenotips/entities/script/PrimaryEntityResolverScriptService.java#L63

Reply:
Fixed.